lldb/tools/lldb-dap/src-ts/extension.ts

```diff
@@ -17,10 +18,32 @@ function createDefaultLLDBDapOptions(): LLDBDapOptions {
       const path = vscode.workspace
         .getConfiguration("lldb-dap", session.workspaceFolder)
         .get<string>("executable-path");
-      if (path) {
-        return new vscode.DebugAdapterExecutable(path, []);
+
+      if (!path) {
+        return packageJSONExecutable;
+      }
+
+      try {
+        const fileStats = await fs.stat(path);
+        if (!fileStats.isFile() && !fileStats.isSymbolicLink()) {
+          throw new Error(`Error: ${path} is not a file`);
+        }
+      } catch (err) {
+        const error: Error = err as Error;
+        const openSettingsAction = "Open Settings";
+        const callBackValue = await vscode.window.showErrorMessage(
+          error.message,
+          { modal: true },
+          openSettingsAction,
+        );
+        if (openSettingsAction === callBackValue) {
+          vscode.commands.executeCommand(
+            "workbench.action.openSettings",
+            "lldb-dap.executable-path",
+          );
+        }
       }
-      return packageJSONExecutable;
+      return new vscode.DebugAdapterExecutable(path, []);
```

Comment:
instead of making a try catch block here, why don't you just show the pop up right away without throwing an error?

Reply:
the first one is when there is a directory / device/ socket (any thing that is not an actual file) that exists in that path 

the second one is when there is nothing in that path

Reply:
> instead of making a try catch block here, why don't you just show the pop up right away without throwing an error?

I would have to show it twice one for when the path does not exist and the other for when it exists and it is not a file